File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& a) {  
        int n = a.size();  
        vector<int> temp;
        for (int i = 0; i < n; i++) {
            if (a[i] != 0) {
                temp.push_back(a[i]);
            }
        }
        int nz = temp.size();
        for (int i = 0; i < nz; i++) {
            a[i] = temp[i];
        }
        for (int i = nz; i < n; i++) {
            a[i] = 0;
        }
    }
};

File: 283-move-zeroes/move-zeroes.rs
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
         let n = nums.len();
         let mut j = -1;
         for i in 0..n {
            if nums[i] == 0 {
                j = i as i32;
                break;
            }
        }
         if j == -1 {
            return;
        }
        for i in (j + 1) as usize..n {
            if nums[i] != 0 {
                nums.swap(i, j as usize);
                j += 1;
            }
        }
    }
}